federation/pkg/federation-controller/namespace/namespace_controller.go

```diff
@@ -329,73 +442,79 @@ func (nc *NamespaceController) delete(namespace *api_v1.Namespace) error {
 			Phase: api_v1.NamespaceTerminating,
 		},
 	}
+	var err error
 	if namespace.Status.Phase != api_v1.NamespaceTerminating {
 		nc.eventRecorder.Event(namespace, api.EventTypeNormal, "DeleteNamespace", fmt.Sprintf("Marking for deletion"))
-		_, err := nc.federatedApiClient.Core().Namespaces().Update(updatedNamespace)
+		_, err = nc.federatedApiClient.Core().Namespaces().Update(updatedNamespace)
 		if err != nil {
 			return fmt.Errorf("failed to update namespace: %v", err)
 		}
 	}
 
-	// Right now there is just 5 types of objects: ReplicaSet, Secret, Ingress, Events and Service.
+	if nc.hasFinalizerFuncInSpec(updatedNamespace, api_v1.FinalizerKubernetes) {
+		// Delete resources in this namespace.
+		updatedNamespace, err = nc.removeKubernetesFinalizer(updatedNamespace)
```

Comment:
To clarify, I named it removeKubernetesFinalizer because it removes the finalizer called FinalizerKubernetes.

> Not sure I follow this part. What kubernetes code are you referring to?

kubernetes namespace controller also has the same code that deletes all resources in a namespace before removing the finalizer - same as what this function does.